excellent/evaluator_test.go

```diff
@@ -53,6 +41,11 @@ func (v *testXObject) Resolve(key string) types.XValue {
 	}
 }
 
+// ToXJSON is called when this type is passed to @(to_json(...))
+func (v *testXObject) ToXJSON() types.XString {
+	return types.ResolveKeys(v, "foo", "bar").ToXJSON()
```

Comment:
Using this to generate the XJSON ensures that each included field has the exact same value as what you'd get if you JSONified that individual field.